File: 30DaysOfCode/day10_binaryNumbers/binaryNumbers.js
process.stdin.resume();
process.stdin.setEncoding('ascii');

var input_stdin = "";
var input_stdin_array = "";
var input_currentline = 0;

process.stdin.on('data', function (data) {
    input_stdin += data;
});

process.stdin.on('end', function () {
    input_stdin_array = input_stdin.split("\n");
    main();    
});

function readLine() {
    return input_stdin_array[input_currentline++];
}

function main() {
    var n = parseInt(readLine());
    
    var max = 0;
    var count = 0;
    
    while (n > 0) {
        if (n % 2 == 1) {
            count++;
            if (count > max) {
                max = count;
            }
        } else {
            count = 0;
        }
        
        n = parseInt(n / 2);
    }
    
    console.log(max);
}

File: 30DaysOfCode/day10_binaryNumbers/binaryNumbers.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
class Solution {

    static void Main(String[] args) {
        int n = Convert.ToInt32(Console.ReadLine());
        
        int max = 0;
        int count = 0;
        
        while (n > 0) {
            if (n % 2 == 1) {
                count++;
                if (count > max) {
                    max = count;
                }
            } else {
                count = 0;
            }
            
            n /= 2;
        }
        
        Console.WriteLine(max);
    }
}
